### PR TITLE
Tasks refactor

### LINKED ISSUE #298: Changes regards to Achievements
- Achievement got renamed to Tasks
  - AchievementCategory -> TaskCategory
  - /api/achievement/** -> /api/task/**
  - achievements -> tasks
  - achievement -> task
  - achievementId -> taskId
  - achievementName -> taskName
  - INVALID_ACHIEVEMENT_ID -> INVALID_TASK_ID
  - *achievement* -> *task*

- TaskCategory: new 'type' property 
  - REGULAR = the same as before, 
  - PROFILE_REQUIRED = not shown on the tasks list but can be accessed throgh the profile menu (this one is for CV-s and profile pictures for GTB)
  
- TaskEntity: new 'format' property
  - NONE = no textarea and file upload
  - TEXT = simple textarea
  - CODE = a basic syntax highlighter (C,C++,C#,Java,Python,plaintext) 
  - FORM = A form that outputs a string in the background. Format: [{'title':'','type':'number|text|textarea','suffix':''}] and for example it outputs sg. like that: "Title1: Value suffix\nTitle2: Value suffix\n"
  
- TaskEntity: new 'formatDescriptor' property: the json if the 'format' is FORM 

- TaskEtity: 'type' property new possible value: 'ONLY_PDF'
  - Add pdf file selector
  - TaskSubmissionStatus new possible value 'INVALID_PDF' 

related to #268 #297

### PR DESCRIPTION
**Pull request by Tschonti**
~~Code editor is WIP, the rest is more or less ready~~

Everything should be ready now. The package of the code editor gives a webpack warning for some reason, but it works fine. It should be investigated in a future issue.

Closes #298, #325

### CODE REVIEW

File: frontend/src/pages/task/taskCategory.page.tsx [hunk 1]
```diff
@@ -1,82 +1,69 @@
...
+                </Link>
+              </Box>
+            ))}
+          </VStack>
+        ) : (
+          <Text>Nincs egyetlen feladat se ebben a kategóriában.</Text>
+        )}
+      </CmschPage>
+    )
+  } else {
```
Comment (DannySS123) [inline frontend/src/pages/task/taskCategory.page.tsx:60]: Error handling missing

File: frontend/src/pages/task/taskCategoryList.page.tsx [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+                </Link>
+              </Box>
+            ))}
+          </VStack>
+        ) : (
+          <Text>Nincs egyetlen feladat se.</Text>
+        )}
+      </CmschPage>
+    )
+  } else {
```
Comment (DannySS123) [inline frontend/src/pages/task/taskCategoryList.page.tsx:50]: Error handling missing

**Review by DannySS123 — CHANGES_REQUESTED** [review]
Great work! 💯

**Comment by berenteb** [comment]
Going great, just noticed these minor change requests, can be merged after...

**Review by berenteb — CHANGES_REQUESTED** [review]
Route name to task and it is good to go

**Review by berenteb — APPROVED** [review]

**Pull request merged**